src/libs/AxelarQueryAPI.ts

```diff
@@ -309,30 +390,41 @@ export class AxelarQueryAPI {
       sourceToken.decimals
     );
 
-    const executionFee = destGasFeeWei.gt(minDestGasFeeWei)
+    const excludedL1ExecutionFee = destGasFeeWei.gt(minDestGasFeeWei)
       ? srcGasFeeWei
       : srcGasFeeWei.mul(minDestGasFeeWei).div(destGasFeeWei);
 
     const actualGasMultiplier = gasMultiplier === "auto" ? executeGasMultiplier : gasMultiplier;
 
-    const executionFeeWithMultiplier =
+    const excludedL1ExecutionFeeWithMultiplier =
       actualGasMultiplier > 1
-        ? executionFee.mul(actualGasMultiplier * 10000).div(10000)
-        : executionFee;
+        ? excludedL1ExecutionFee.mul(actualGasMultiplier * 10000).div(10000)
+        : excludedL1ExecutionFee;
+
+    const [l1ExecutionFee, l1ExecutionFeeWithMultiplier] = await this.calculateL1FeeForDestL2(
+      destinationChainId,
+      destToken,
+      executeData,
+      sourceToken,
+      ethereumToken,
+      actualGasMultiplier
+    );
 
     return gmpParams?.showDetailedFees
       ? {
           baseFee,
           expressFee,
-          executionFee: executionFee.toString(),
-          executionFeeWithMultiplier: executionFeeWithMultiplier.toString(),
+          executionFee: excludedL1ExecutionFeeWithMultiplier.toString(),
+          executionFeeWithMultiplier: excludedL1ExecutionFeeWithMultiplier.toString(),
+          l1ExecutionFeeWithMultiplier: l1ExecutionFeeWithMultiplier.toString(),
+          l1ExecutionFee: l1ExecutionFee.toString(),
           gasLimit,
           gasMultiplier: actualGasMultiplier,
           minGasPrice: minGasPrice === "0" ? "NA" : minGasPrice,
           apiResponse,
           isExpressSupported: expressSupported,
         }
-      : executionFeeWithMultiplier.add(baseFee).toString();
+      : l1ExecutionFeeWithMultiplier.add(excludedL1ExecutionFee).add(baseFee).toString();
```

Comment:
shouldn't this add `excludedL1ExecutionFeeWithMultiplier`?

Reply:
good catch!